src/main/java/com/commercetools/project/sync/category/CategorySyncer.java

```diff
@@ -0,0 +1,61 @@
+package com.commercetools.project.sync.category;
+
+import static com.commercetools.sync.categories.utils.CategoryReferenceReplacementUtils.buildCategoryQuery;
+import static com.commercetools.sync.categories.utils.CategoryReferenceReplacementUtils.replaceCategoriesReferenceIdsWithKeys;
+
+import com.commercetools.project.sync.Syncer;
+import com.commercetools.sync.categories.CategorySync;
+import com.commercetools.sync.categories.CategorySyncOptions;
+import com.commercetools.sync.categories.CategorySyncOptionsBuilder;
+import com.commercetools.sync.categories.helpers.CategorySyncStatistics;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import io.sphere.sdk.categories.queries.CategoryQuery;
+import io.sphere.sdk.client.SphereClient;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CategorySyncer
+    extends Syncer<
+        Category,
+        CategoryDraft,
+        CategorySyncStatistics,
+        CategorySyncOptions,
+        CategoryQuery,
+        CategorySync> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CategorySyncer.class);
+
+  /** Instantiates a {@link Syncer} instance. */
+  private CategorySyncer(
+      @Nonnull final CategorySync categorySync,
+      @Nonnull final CategoryQuery categoryQuery,
+      @Nonnull final SphereClient sourceClient,
+      @Nonnull final SphereClient targetClient) {
+    super(categorySync, categoryQuery, sourceClient, targetClient);
+    // TODO: Instead of reference expansion, we could cache all keys and replace references
```

Comment:
Where do you have expansion in this class?

Reply:
inside `buildCategoryQuery()` here: https://github.com/commercetools/commercetools-project-sync/blob/18058317d5aae7f612ad9a772b03c1187b6c9432/src/main/java/com/commercetools/project/sync/category/CategorySyncer.java#L53